### PR TITLE
Improve errors in the 'export with Signal' feature.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/MessagesViewController.m [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ - (void)initializeCollectionViewLayout
 {
     [self.collectionView.collectionViewLayout setMessageBubbleFont:[UIFont ows_dynamicTypeBodyFont]];
 
-    self.collectionView.showsVerticalScrollIndicator = NO;
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/MessagesViewController.m]: Just sneaking this in? Bold move! 👍

File: Signal/src/AppDelegate.m [hunk 2]
```diff
@@ -268,30 +268,64 @@ - (BOOL)application:(UIApplication *)application
...
+                          @"Title for the alert indicating the 'export with signal' attachment had an error.")
+                                  message:NSLocalizedString(@"EXPORT_WITH_SIGNAL_ERROR_MESSAGE_INVALID_FILENAME",
+                                              @"Message for the alert indicating the 'export with signal' file had an "
+                                              @"invalid filename.")];
             return NO;
         }
         NSString *fileExtension = [filename pathExtension];
         if (fileExtension.length < 1) {
             DDLogError(@"Application opened with URL missing file extension: %@", url);
+            [self showErrorAlertWithTitle:
```
Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: Having an error is better than failing silently, but we actually want to support this case right?

File: Signal/src/AppDelegate.m [hunk 3]
```diff
@@ -268,30 +268,64 @@ - (BOOL)application:(UIApplication *)application
...
+                          @"Title for the alert indicating the 'export with signal' attachment had an error.")
+                                  message:NSLocalizedString(@"EXPORT_WITH_SIGNAL_ERROR_MESSAGE_UNKNOWN_TYPE",
+                                              @"Message for the alert indicating the 'export with signal' file had "
+                                              @"unknown type.")];
             return NO;
         }
         NSString *utiType = [MIMETypeUtil utiTypeForFileExtension:fileExtension];
         if (utiType.length < 1) {
             DDLogError(@"Application opened with URL of unknown UTI type: %@", url);
+            [self showErrorAlertWithTitle:
```
Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: Having an error is better than failing silently, but we actually want to support this case right?

File: Signal/src/ViewControllers/MessagesViewController.m (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/MessagesViewController.m]: Sorry that was for debugging. Will remove.

File: Signal/src/AppDelegate.m (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: Unfortunately "open with Signal" isn't presented for files without a file extension - or for files of unknown file extension.

File: Signal/src/AppDelegate.m (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: Unfortunately "open with Signal" isn't presented for files without a file extension - or for files of unknown file extension.

**Pull request merged**